presto-tpch/src/main/java/com/facebook/presto/tpch/StatisticsEstimator.java

```diff
@@ -0,0 +1,258 @@
+        }
+        else {
+            return rescale(big, small, scaleFactor);
+        }
+    }
+
+    private ColumnStatisticsData rescale(ColumnStatisticsData big, ColumnStatisticsData small, double scaleFactor)
+    {
+        if (areClose(big, small)) {
+            return new ColumnStatisticsData(
+                    big.getUniqueValuesCount(),
+                    big.getNullsCount(),
+                    checkPresent(withBoth(big.getLowestValue(), small.getLowestValue(), this::checkClose)),
+                    checkPresent(withBoth(big.getHighestValue(), small.getHighestValue(), this::checkClose))
+            );
+        }
+        else {
+            Function<Number, Double> scale = value -> value.doubleValue() * scaleFactor;
+            return new ColumnStatisticsData(
+                    big.getUniqueValuesCount().map(scale.andThen(Number::longValue)),
+                    big.getNullsCount().map(scale.andThen(Number::longValue)),
+                    Types.tryCast(big.getLowestValue(), Number.class),
+                    Types.tryCast(big.getHighestValue(), Number.class).map(scale));
+        }
+    }
+
+    private boolean areClose(ColumnStatisticsData big, ColumnStatisticsData small)
+    {
+        Optional<Long> smallCount = small.getUniqueValuesCount();
+        Optional<Long> bigCount = big.getUniqueValuesCount();
+
+        return smallCount.isPresent() == bigCount.isPresent() &&
+                withBoth(smallCount, bigCount, this::areClose)
+                        .orElse(boundsAreClose(big, small));
+    }
+
+    private Boolean boundsAreClose(ColumnStatisticsData big, ColumnStatisticsData small)
+    {
+        return withBoth(small.getLowestValue(), big.getLowestValue(), this::areClose)
+                .flatMap(result -> withBoth(small.getHighestValue(), big.getHighestValue(), this::areClose))
+                .orElse(false);
+    }
+
+    private Object checkClose(Object leftValue, Object rightValue)
+    {
+        checkArgument(
+                areClose(leftValue, rightValue),
+                format("Values must be close to each other, got [%s] and [%s]", leftValue, rightValue));
+        return leftValue;
+    }
+
+    private boolean areClose(Object leftValue, Object rightValue)
+    {
+        checkSameType(leftValue, rightValue);
+        if (leftValue instanceof String) {
+            return leftValue.equals(rightValue);
+        }
+        else {
+            Number left = checkType(leftValue, Number.class);
+            Number right = checkType(rightValue, Number.class);
+            return areClose(left.doubleValue(), right.doubleValue());
+        }
+    }
+
+    private boolean areClose(double smallCount, double bigCount)
+    {
+        double difference = abs(bigCount - smallCount);
+        return difference <= abs(smallCount) * 0.1 || difference <= 0;
+    }
+
+    private TableStatisticsData readStatistics(String tableName, Predicate<Map<TpchColumn<?>, Object>> constraint, String scaleFactor)
```

Comment:
Thought about this too, but there's significant logic of approximating stats for partitioned tables and it felt right to leave it here.